### PR TITLE
Navbar styling rework

### PR DESCRIPTION
**Pull request by silverwind**
- Extract navbar CSS to own file
- Reduce height from 52px to 50px
- Give every item a hover effect of of 36px, including the logo and on mobile
- Consistent horizontal padding of 10px left and right

Screenshot 2023-06-18 at 13 41 16

Screenshot 2023-06-18 at 14 03 43
Screenshot 2023-06-18 at 14 03 18

Screenshot 2023-06-18 at 13 58 28

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
@wxiaoguang care to review?

**Comment by wxiaoguang** [comment]
I do not understand the layout ".menu > .navbar-left > .item", it doesn't seems a correct Fomantic UI layout.

If it wants to use Fomantic Menu, then I think it's better to follow Fomantic UI's examples. If it doesn't want to, then do not use "menu"

**Comment by silverwind** [comment]
We can likely remove the fomantic menu, but it'll require a bit of more CSS as stuff like `.active` link styling relies on it.

**Review by wxiaoguang — APPROVED** [review]
It seems fragile to do ".menu > .navabr-left > .item" and ".menu > .navabr-right > .item"

If it needs to re-use Fomantic styles, it could be:

```
<nav>
   <div class="navbar-left ui menu">
      <div class=item></div>
   </div>
   <div class="navbar-right ui menu">
      <div class=item></div>
   </div>
</nav>
```

Then it doesn't violate Fomantic UI structures.

While I won't block it.

**Comment by silverwind** [comment]
I will have a look whether this can be cleanup up easily.

**Comment by silverwind** [comment]
Wow this turns out too be way too tricky because of how much of a hack this mobile menu is.

**Comment by silverwind** [comment]
> It seems fragile to do ".menu > .navabr-left > .item" and ".menu > .navabr-right > .item"
> 
> If it needs to re-use Fomantic styles, it could be:
> 
> ```
> <nav>
>    <div class="navbar-left ui menu">
>       <div class=item></div>
>    </div>
>    <div class="navbar-right ui menu">
>       <div class=item></div>
>    </div>
> </nav>
> ```
> 
> Then it doesn't violate Fomantic UI structures.
> 
> While I won't block it.

Implemented this now, it took some effort with making mobile menu `flex-wrap` and the first two items 50% width, but it seems to work fine now.

File: templates/base/head_navbar.tmpl [hunk 1]
```diff
@@ -3,62 +3,63 @@
...
+		<a class="item" id="navbar-logo" href="{{AppSubUrl}}/" aria-label="{{if .IsSigned}}{{.locale.Tr "dashboard"}}{{else}}{{.locale.Tr "home"}}{{end}}">
+			<img width="30" height="30" src="{{AssetUrlPrefix}}/img/logo.svg" alt="{{.locale.Tr "logo"}}" aria-hidden="true"/>
 		</a>
 
 		<!-- mobile right menu, it must be here because in mobile view, each item is a flex column, the first item is a full row column -->
-		<div class="ui secondary menu navbar-mobile-right gt-gap-2">
+		<div class="ui secondary menu item navbar-mobile-right gt-gap-2">
 			{{if .IsSigned}}
-			<a class="item gt-mx-0 gt-p-3" href="{{AppSubUrl}}/notifications" data-tooltip-content="{{.locale.Tr "notifications"}}" aria-label="{{.locale.Tr "notifications"}}">
+			<a class="item mobile-right-item mobile-notifications-item gt-mr-2 gt-p-3" href="{{AppSubUrl}}/notifications" data-tooltip-content="{{.locale.Tr "notifications"}}" aria-label="{{.locale.Tr "notifications"}}">
```
Comment (wxiaoguang) [inline templates/base/head_navbar.tmpl]: `mobile-notifications-item` is no-op ?

File: templates/base/head_navbar.tmpl [hunk 2]
```diff
@@ -3,62 +3,63 @@
...
 
-<nav id="navbar" class="ui secondary stackable menu" aria-label="{{.locale.Tr "aria.navbar"}}">
-	<div class="item">
+<nav id="navbar" aria-label="{{.locale.Tr "aria.navbar"}}">
+	<div class="navbar-left ui secondary menu">
 		<!-- the logo -->
-		<a href="{{AppSubUrl}}/" aria-label="{{if .IsSigned}}{{.locale.Tr "dashboard"}}{{else}}{{.locale.Tr "home"}}{{end}}">
-			<img width="30" height="30" src="{{AssetUrlPrefix}}/img/logo.svg" alt="{{.locale.Tr "logo"}}" aria-hidden="true">
+		<a class="item" id="navbar-logo" href="{{AppSubUrl}}/" aria-label="{{if .IsSigned}}{{.locale.Tr "dashboard"}}{{else}}{{.locale.Tr "home"}}{{end}}">
+			<img width="30" height="30" src="{{AssetUrlPrefix}}/img/logo.svg" alt="{{.locale.Tr "logo"}}" aria-hidden="true"/>
```
Comment (wxiaoguang) [inline templates/base/head_navbar.tmpl]:
```suggestion
			<img width="30" height="30" src="{{AssetUrlPrefix}}/img/logo.svg" alt="{{.locale.Tr "logo"}}" aria-hidden="true">
```

**Review by wxiaoguang — APPROVED** [review]

File: templates/base/head_navbar.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/base/head_navbar.tmpl]: Yeah, a leftover from previous attempts, removed.

**Comment by silverwind** [comment]
There's some regression in the dropdown menus, fixing...

**Comment by silverwind** [comment]
All fixed. These nested `.item` were the problem.

**Pull request merged**